### DISCUSSION TITLE
What happens if useEffect is registering an input multiple times?

### DISCUSSION THREAD

**Discussion by mauricioaznar** [react-hook-form/react-hook-form#2719] [Q&A]
So I have a useFieldArray 

```
const expenseItems = useFieldArray(
   {
     control,
     name: "expense_items"
   }
);

```
Each item has a subtotal property, which I am using to calculate a total.

`const watchExpenseItems = watch('expense_items')`

```
let total = watchExpenseItems.reduce((acc, expenseItem) => {
   return expenseItem.subtotal !== '' ? acc + Number(expenseItem.subtotal) : acc
 }, 0)

```
```
useEffect(() => {
   register({name: "_total", type: "custom"},
     {
        validate: () => {
          return total > 100 ? true : 'Total is less than 100'
       }
     });
}, [total]);
```

Of course the code is not complete. I was just wondering what happens when total gets recalculated. Does the register function overwrite the previous '_total' inside of the useEffect. Is there some issue having this implementation?. 

Thank you in advance and congratulations, the repo is awesome!.

**Comment by bluebill1049** [comment]
hmmm, the custom register is not working in useFieldArray at the moment. why do you need to re-register your input again? just curious.

**Reply by mauricioaznar** [reply]
Hello bill!
Sorry I forgot to erase the "type: custom" property. I was just trying it out but I dont think it matters for the purpose of my question.
